src/Moq/StringBuilderExtensions.cs

```diff
@@ -41,18 +41,31 @@ public static StringBuilder AppendNameOf(this StringBuilder stringBuilder, Metho
 
 			if (includeGenericArgumentList && method.IsGenericMethod)
 			{
-				stringBuilder.Append('<');
-				var genericArguments = method.GetGenericArguments();
-				for (int i = 0, n = genericArguments.Length; i < n; ++i)
-				{
-					if (i > 0)
-					{
-						stringBuilder.Append(", ");
-					}
-					stringBuilder.AppendNameOf(genericArguments[i]);
-				}
+				stringBuilder.AppendGenericArguments(method);
+			}
+
+			return stringBuilder;
+		}
+
+		public static StringBuilder AppendNameOfAddEvent(this StringBuilder stringBuilder, MethodBase method, bool includeGenericArgumentList)
+		{
+			stringBuilder.Append(method.Name.Replace("add_", "")).Append(" += ");
```

Comment:
👍

Reply:
Oh I just realised we could have even removed `.Substring()`, since `StringBuilder.Append` gives you the means to append parts of a strings. (But never mind, this part of the code base isn't performance critical, and there are tons of other things that could be optimized there... this is good enough for the moment!)